neo/io/nestio.py

```diff
@@ -134,53 +134,62 @@ def __read_analogsignals(self, gid_list, time_unit, t_start=None,
                                                             gid_list,
                                                             t_start,
                                                             t_stop)
-        # loading raw data columns
-        data = self.avail_IOs['dat'].get_columns(
-            column_ids=column_ids,
-            condition=condition,
-            condition_column=condition_column,
-            sorting_columns=sorting_column)
-
-        sampling_period = self._check_input_sampling_period(sampling_period,
-                                                            time_column,
-                                                            time_unit,
-                                                            data)
-        analogsignal_list = []
 
-        # extracting complete gid list for anasig generation
-        if (gid_list == []) and id_column is not None:
-            gid_list = np.unique(data[:, id_column])
+        analogsignal_list = []
```

Comment:
If I understand the NEST analogsignal outputs correctly, one could consider creating one `AnalogSignal` per file (i.e., for each entry in `self.IOs`), and then add all channels (gids) into columns of that `AnalogSignal`. Currently, separate `AnalogSignal`s are created for each gid in each file.

Reply:
This probably needs discussion.

Reply:
I agree that a single analogsignal object should be created per source file